src/TSTransformer/nodes/class/transformClassLikeDeclaration.ts

```diff
@@ -354,57 +355,69 @@ export function transformClassLikeDeclaration(state: TransformState, node: ts.Cl
-			methods.push(member);
+			const method = member;
+			if (ts.isIdentifier(method.name) || ts.isStringLiteral(method.name)) {
+				if (luau.isMetamethod(method.name.text)) {
+					DiagnosticService.addDiagnostic(errors.noClassMetamethods(method.name));
+				}
+
+				if (!!ts.getSelectedSyntacticModifierFlags(method, ts.ModifierFlags.Static)) {
+					if (instanceType.getProperty(method.name.text) !== undefined) {
+						DiagnosticService.addDiagnostic(errors.noInstanceMethodCollisions(method));
+					}
+				} else {
+					if (classType.getProperty(method.name.text) !== undefined) {
+						DiagnosticService.addDiagnostic(errors.noStaticMethodCollisions(method));
+					}
+				}
+			}
+
+			luau.list.pushList(
+				statementsInner,
+				transformMethodDeclaration(state, method, { name: "name", value: internalName }),
+			);
 		} else if (ts.isPropertyDeclaration(member)) {
-			// do not emit non-static properties here
-			if (!ts.hasStaticModifier(member)) {
-				continue;
+			// Non-static properties are handled in constructor
+			if (ts.hasStaticModifier(member)) {
+				luau.list.pushList(statementsInner, transformPropertyDeclaration(state, member, internalName));
 			}
-			staticProperties.push(member);
+		} else if (ts.isClassStaticBlockDeclaration(member)) {
+			const staticBlockStatements = luau.list.make<luau.Statement>(
+				luau.create(luau.SyntaxKind.VariableDeclaration, {
+					left: luau.globals.self,
+					right: internalName,
+				}),
+			);
+			luau.list.pushList(staticBlockStatements, transformStatementList(state, member.body.statements));
+			luau.list.push(
+				statementsInner,
+				luau.create(luau.SyntaxKind.DoStatement, {
+					statements: staticBlockStatements,
+				}),
+			);
 		} else if (ts.isAccessor(member)) {
 			DiagnosticService.addDiagnostic(errors.noGetterSetter(member));
 		} else {
 			assert(false, `Class member type not implemented: ${getKindName(member.kind)}`);
 		}
 	}
 
-	const classType = state.typeChecker.getTypeOfSymbolAtLocation(node.symbol, node);
-	const instanceType = state.typeChecker.getDeclaredTypeOfSymbol(node.symbol);
-
-	for (const method of methods) {
-		if (ts.isIdentifier(method.name) || ts.isStringLiteral(method.name)) {
-			if (luau.isMetamethod(method.name.text)) {
-				DiagnosticService.addDiagnostic(errors.noClassMetamethods(method.name));
-			}
-
-			if (!!ts.getSelectedSyntacticModifierFlags(method, ts.ModifierFlags.Static)) {
-				if (instanceType.getProperty(method.name.text) !== undefined) {
-					DiagnosticService.addDiagnostic(errors.noInstanceMethodCollisions(method));
-				}
-			} else {
-				if (classType.getProperty(method.name.text) !== undefined) {
-					DiagnosticService.addDiagnostic(errors.noStaticMethodCollisions(method));
-				}
-			}
-		}
-
-		luau.list.pushList(
-			statementsInner,
-			transformMethodDeclaration(state, method, { name: "name", value: internalName }),
-		);
-	}
-
+	// TODO: Make this more generic, and add it for `[Symbol.iterator]` => `__iter`
```

Comment:
Not 100% sure I agree with this idea. Please remove this comment for now and make an issue proposal with design ideas.

Reply:
Will do